File: strand-complement/complement.py
strand = input()
inverseStrand = strand[::-1]
strandComplement = ""


for base in inverseStrand:
    if base == 'A':
        strandComplement += "T"
    elif base == "T":
        strandComplement += "A"
    elif base == "G":
        strandComplement += "C"
    elif base == "C":
        strandComplement += "G"

print(strandComplement)

File: strand-complement/complement.cpp
#include <iostream> 

using namespace std; 

int main(){
    string strand, strandComplement = "";

    cin >> strand;

    for(int i = strand.length() - 1; i >= 0; i--){
        if(strand.at(i) == 'A'){
            strandComplement.append("T");
        } else if(strand.at(i) == 'T'){
            strandComplement.append("A");
        } else if(strand.at(i) == 'C'){
            strandComplement.append("G");
        } else if(strand.at(i) == 'G'){
            strandComplement.append("C");
        }
    }

    cout << strandComplement;

    return 0;
}